matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncResponseHandler.kt

```diff
@@ -70,127 +71,165 @@ internal class SyncResponseHandler @Inject constructor(
+                Timber.v("Finish handling rooms in $it ms")
             }
+        }
+    }
 
-            // "crypto_sync_handler_onSyncCompleted" span
-            measureSpan(relevantPlugins, "task", "crypto_sync_handler_onSyncCompleted") {
-                measureTimeMillis {
-                    cryptoSyncHandler.onSyncCompleted(syncResponse)
-                }.also {
-                    Timber.v("cryptoSyncHandler.onSyncCompleted took $it ms")
+    private fun handleAccountData(reporter: ProgressReporter?, realm: Realm, syncResponse: SyncResponse) {
+        // Child "handle_account_data" span
+        relevantPlugins.measureSpan("task", "handle_account_data") {
+            measureTimeMillis {
+                reportSubtask(reporter, InitialSyncStep.ImportingAccountData, 1, 0.1f) {
+                    Timber.v("Handle accountData")
+                    userAccountDataSyncHandler.handle(realm, syncResponse.accountData)
                 }
+            }.also {
+                Timber.v("Finish handling accountData in $it ms")
+            }
+        }
+    }
+
+    private fun handlePresence(realm: Realm, syncResponse: SyncResponse) {
+        // Child "handle_presence" span
+        relevantPlugins.measureSpan("task", "handle_presence") {
+            measureTimeMillis {
+                Timber.v("Handle Presence")
+                presenceSyncHandler.handle(realm, syncResponse.presence)
+            }.also {
+                Timber.v("Finish handling Presence in $it ms")
             }
+        }
+    }
 
-            // post sync stuffs
-            monarchy.writeAsync {
-                roomSyncHandler.postSyncSpaceHierarchyHandle(it)
+    private suspend fun aggregateSyncResponse(aggregator: SyncResponsePostTreatmentAggregator) {
+        // "aggregator_management" span
+        relevantPlugins.measureSpan("task", "aggregator_management") {
+            // Everything else we need to do outside the transaction
+            measureTimeMillis {
+                aggregatorHandler.handle(aggregator)
+            }.also {
+                Timber.v("Aggregator management took $it ms")
             }
-            Timber.v("On sync completed")
+        }
+    }
+
+    private suspend fun postTreatmentSyncResponse(syncResponse: SyncResponse, isInitialSync: Boolean) {
+        // "sync_response_post_treatment" span
+        relevantPlugins.measureSpan("task", "sync_response_post_treatment") {
+            measureTimeMillis {
+                syncResponse.rooms?.let {
+                    checkPushRules(it, isInitialSync)
+                    userAccountDataSyncHandler.synchronizeWithServerIfNeeded(it.invite)
+                    dispatchInvitedRoom(it)
+                }
+            }.also {
+                Timber.v("SyncResponse.rooms post treatment took $it ms")
+            }
+        }
+    }
+
+    private fun markCryptoSyncCompleted(syncResponse: SyncResponse) {
+        // "crypto_sync_handler_onSyncCompleted" span
```

Comment:
I wonder if it is useful to keep these comments about spans since now we have dedicated private methods?